libpod/networking_linux.go

```diff
@@ -934,6 +943,30 @@ func (c *Container) getContainerNetworkInfo() (*define.InspectNetworkSettings, e
 	return settings, nil
 }
 
+// setupNetworkDescriptions adds networks and eth values to the container's
+// network descriptions
+func (c *Container) setupNetworkDescriptions(networks []string) error {
+	netDescriptions := c.state.NetworkDescriptions
+	origLen := len(netDescriptions)
+	for _, n := range networks {
+		// if the network is not in the map, add it
+		if _, exists := netDescriptions[n]; !exists {
+			if len(c.state.NetworkDescriptions) < 1 {
```

Comment:
This is not an effective check for a nil pointer, it has to happen before the lookup on line 953 to prevent a panic

Reply:
Also, I am really confused as to why we are referring to both c.state.NetworkDescriptions and netDescriptions which is just a pointer to the same map here?